packages/x-charts-pro/src/Heatmap/HeatmapItem.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { useSlotProps } from '@mui/base/utils';
+import composeClasses from '@mui/utils/composeClasses';
+import { useItemHighlighted } from '@mui/x-charts/context';
+import { useInteractionItemProps, SeriesId } from '@mui/x-charts/internals';
+import { HeatmapClasses, getHeatmapUtilityClass } from './heatmapClasses';
+
+export interface HeatmapItemSlots {
+  /**
+   * The component that renders the heatmap cell.
+   * @default HeatmapCell
+   */
+  cell?: React.ElementType;
+}
+
+export interface HeatmapItemSlotProps {
+  cell?: Partial<React.ComponentPropsWithRef<'rect'>>;
+}
+
+export interface HeatmapItemProps {
+  dataIndex: number;
+  seriesId: SeriesId;
+  value: number;
+  width: number;
+  height: number;
+  x: number;
+  y: number;
+  color: string;
+  /**
+   * The props used for each component slot.
+   * @default {}
+   */
+  slotProps?: HeatmapItemSlotProps;
+  /**
+   * Overridable component slots.
+   * @default {}
+   */
+  slots?: HeatmapItemSlots;
+}
+
+export interface HeatmapItemOwnerState {
+  seriesId: SeriesId;
+  dataIndex: number;
+  color: string;
+  isFaded: boolean;
+  isHighlighted: boolean;
+  classes?: Partial<HeatmapClasses>;
+}
+
+const HeatmapCell = styled('rect', {
+  name: 'MuiHeatmap',
+  slot: 'Cell',
+  overridesResolver: (_, styles) => styles.arc,
+})<{ ownerState: HeatmapItemOwnerState }>(({ ownerState }) => ({
+  filter:
+    (ownerState.isHighlighted && 'saturate(120%)') ||
+    (ownerState.isFaded && 'saturate(80%)') ||
+    undefined,
+  fill: ownerState.color,
+  shapeRendering: 'crispEdges',
+}));
```

Comment:
Should we use the actual classes for the css styles? This way the css is more atomic, and the "highlight/faded" styles don't require a new "root" style.

```suggestion
const HeatmapCell = styled('rect', {
  name: 'MuiHeatmap',
  slot: 'Cell',
  overridesResolver: (_, styles) => [
    { [`&.${heatmapClasses.faded}`]: styles.faded },
    { [`&.${heatmapClasses.highlighted}`]: styles.highlighted },
    styles.cell,
  ],
})<{ ownerState: HeatmapItemOwnerState }>(({ ownerState }) => ({
  fill: ownerState.color,
  shapeRendering: 'crispEdges',
  [`&.${heatmapClasses.faded}`]: {
    filter: 'saturate(80%)',
  },
  [`&.${heatmapClasses.highlighted}`]: {
    filter: 'saturate(120%)',
  },
}));
```

https://github.com/mui/mui-x/assets/9891233/e8c2886f-2eb5-4bb2-ae4f-7a6c78419ad3

https://github.com/mui/mui-x/assets/9891233/af46128c-60c6-4404-ae3a-ec269f37f066

Reply:
The main advantage of the current solution is the use of a single classe as a CSS selector. Which means you only need two classes to be more specific and override the styling

Reply:
This feels a bit odd, but I understand the rationale.